Documentation/README.md

```diff
@@ -57,3 +57,130 @@ tar -xf <tarball-destination> -C <path-to-extraction-root>
+
+Let's say that you want to produce a runtime that has a new API that it's implementation lives in coreclr and you want to create a .NET Core console app that uses it to test it without waiting for a new SDK or Runtime to be produced by official builds. Imagine we will add an API that just throws in Hashtable. 
+
+Hashtable implementation lives in coreclr and it's APIs are exposed in corefx in the System.Runtime.Extensions.cs contract. So in my local coreclr repo I will just add Hashtable.cs the following API:
+
+```cs
+public static void PNSE()
+{
+    throw new PlatformNotSupportedException();
+}
+```
+
+Then in my local corefx repo let's expose it in System.Runtime.Extensions.cs in the Hashtable class:
+
+```cs
+public static void PNSE() { }
+```
+
+Of course we need to commit this changes in our current repos. Once we've commited this locally, let's move to the source-build repo.
+
+Once in the source-build in our commmand prompt or terminal, lets navigate to `src/coreclr`. Once there, we will need to add a remote to the local coreclr repo, in example:
+```console
+git remote add local C:\repos\coreclr
+```
+
+Then, let's fetch the changes from our repo, by running:
+```console
+git fetch local
+```
+
+Once we've fetch the changes, we just need to checkout the branch where we made the changes:
+```console
+git checkout branchWithChanges
+```
+
+Just repeat this steps under `src/corefx` as well, to pull the corefx changes. Once we've done that, let's navigate to the `source-build` root and run:
+
+```console
+build.{cmd|sh} /p:RootRepo=core-setup
+```
+
+Note that since you've changed the coreclr and corefx commits, the incremental build will find the current commit changed so if you don't want to see messages like:
+```console
+WARNING: submodule src/corefx, currently at 831264e53e5b9333850baa659af8a2857a9cb9b7, has diverged from checked-in
+version 5b7674e4ae5cc782e99f50b2919dfdeb29106a46
+If you are changing a submodule branch or moving a submodule backwards, this is expected.
+Should I checkout src/corefx to the expected commit 5b7674e4ae5cc782e99f50b2919dfdeb29106a46 [N]o / [y]es / [q]uit
+```
+
+After you've pulled the changes, before you build, in your `source-build` repo just add the changes with, `git add src/corefx` and/or `git add src/coreclr`.
+
+Once the build is done, we will have a .NET Runtime containing the `Hashtable.PNSE()` API that can be used with our local cli in .NET Core projects.
+
+### How to use the produced Runtime
+
+The produced runtime will be under `bin/obj/{architecture}/{configuration}/blob-feed/packages`.
+
+Go to that folder and just run the installer. Then create a new .NET Core project:
```

Comment:
Not knowing much about the installers, this seems to me like it might hurt someone down the road when, say, a "real" 2.1.1 is released but they have a custom 2.1.1 installed with their own changes (and they forget they did that). Is it possible/worth documenting a way that doesn't involve installing?

Reply:
Yes I think installing the runtime is only necessary in order to use `dotnet run` but it you publish the app the runtime is not needed to be installed. However I will confirm this and I’ll update accordingly.